src/main/kotlin/com/kryszak/gwatlin/api/ApiLanguage.kt

```diff
@@ -0,0 +1,57 @@
+package com.kryszak.gwatlin.api
+
+import java.util.*
+
+/**
+ * Enum representing the languages supported by the API
+ * @param locale the locale that's backing the language
+ */
+enum class ApiLanguage(val locale: Locale) {
+    /**
+     * English
+     */
+    EN(Locale.ENGLISH),
+
+    /**
+     * German
+     */
+    DE(Locale.GERMAN),
+
+    /**
+     * Spanish
+     */
+    ES(Locale.forLanguageTag("es")),
+
+    /**
+     * French
+     */
+    FR(Locale.FRENCH),
+
+    /**
+     * Chinese
+     */
+    ZH(Locale.CHINESE);
+
+    companion object {
+        private val localeLanguageMap by lazy { values().associateBy { it.locale } }
+        private val stringLanguageMap by lazy { values().associateBy { it.locale.language } }
+
+        /**
+         * @return the language that's corresponding to the supplied locale, or null if no language matches the locale
+         */
+        @JvmStatic
+        fun fromLocale(locale: Locale) = localeLanguageMap[locale]
```

Comment:
Do we need those static methods? Clients should just provide selected value from enum, there's no need for manual parsing from locale/string values

Reply:
My idea here was that it makes interoperability easier. It doesn't really add complexity in my opinion and allows an easy mapping of the common Locale objects that Java provides (or any ISO 639-1 language code respectively) to available API languages. So I wouldn't say it's necessary but it is a nice thing to have in my opinion.

Reply:
That would make sense if any client of API would need to create `ApiLanguage` instance from another object, but in our case he just need to provide any of predefined static values - so IMO those methods are not needed. Client using language selection does not need to know that it uses `Language` class under the hood, right?

Reply:
I don't see a reason why we shouldn't provide this admittedly pretty simple ability. A multilingual application might already use Locale objects and this way they can just translate to our enum without having to implement their own mapping.

A Discord bot for example might offer their results in multiple languages, so they'd need to implement their own mapping from their internal (probably Locale based) to our language Enum.

Reply:
Well, you convinced me